### PR TITLE
improved pageview tracking

### PR DESCRIPTION
**Pull request by jnsdls**

### CODE REVIEW

**Comment by jnsdls** [comment]
#### 📝 Changed routes:


| [**…k\]/\[...customContract\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet]/[network]/[...customContract]) | [**…on-drop/\[edition-drop\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet]/[network]/edition-drop/[edition-drop]) |
|:-:|:-:|
|<a href="https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet]/[network]/[...customContract]">Screenshot of /[wallet]/[network]/[...customContract]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/0Fu3-34SMUEbMQSuRrnTszKBNcA6zI2h5B1C07JsIKV8xwQENtGBQbIb2cu19_P1SnV1pclor16XGcJ1n0WOHw.png">(view full size)</a> |<a href="https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet]/[network]/edition-drop/[edition-drop]">Screenshot of /[wallet]/[network]/edition-drop/[edition-drop]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/Px7W6_LMonKUbNKJquDlpgo7NGM1i2O7ELU2pqr2JPocHlt4qWrPmEJy4Xy8IicgLeQbwpnevazv2iBdWiVoFQ.png">(view full size)</a> |



| [**…ork\]/edition/\[edition\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet]/[network]/edition/[edition]) | [**…ketplace/\[marketplace\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet]/[network]/marketplace/[marketplace]) |
|:-:|:-:|
|<a href="https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet]/[network]/edition/[edition]">Screenshot of /[wallet]/[network]/edition/[edition]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/MJgUvEqYEX5v-_JcbZKrLhSoSzxN2mxTXG_fatSRN5iQBGmTdQxlfFP8kzbUnMEib-palUkYes52iXDc2WLVBA.png">(view full size)</a> |<a href="https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet]/[network]/marketplace/[marketplace]">Screenshot of /[wallet]/[network]/marketplace/[marketplace]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/5sjvxrIRSZ1RHf76hKndbiOpNhQ8_l8TmrblJ-U_yqJKsL8Am0uhNaTKa4KUQlMZ3740Wn9padA_GY_2-Uu7AA.png">(view full size)</a> |



| [**…ction/\[nft-collection\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet]/[network]/nft-collection/[nft-collection]) | [**…k\]/nft-drop/\[nft-drop\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet]/[network]/nft-drop/[nft-drop]) |
|:-:|:-:|
|<a href="https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet]/[network]/nft-collection/[nft-collection]">Screenshot of /[wallet]/[network]/nft-collection/[nft-collection]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/jSrV7KHlGfaLNNPMccsa_5YatFK41RgHIYkuxFqKA1toATZ44-4f937npV4en_17AIJeMiXNwNDlnZxZJFFMdA.png">(view full size)</a> |<a href="https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet]/[network]/nft-drop/[nft-drop]">Screenshot of /[wallet]/[network]/nft-drop/[nft-drop]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/YAS8ZoXzTSzIbta8urYs_S6FSB6Fo8l8z89kqSytZdNtIS-EXRjc7z7nry6mH2-uBXCZPaQLgLnctsOe2wDZPQ.png">(view full size)</a> |



And 22 other routes:
- [**/\[wallet\]/\[network\]/signature-drop/\[signature-drop\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet]/[network]/signature-drop/[signature-drop])
- [**/\[wallet\]/\[network\]/split/\[split\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet]/[network]/split/[split])
- [**/\[wallet\]/\[network\]/token-drop/\[token-drop\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet]/[network]/token-drop/[token-drop])
- [**/\[wallet\]/\[network\]/token/\[token\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet]/[network]/token/[token])
- [**/\[wallet\]/\[network\]/vote/\[vote\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet]/[network]/vote/[vote])
- [**/\[wallet\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/[wallet])
- [**/api/gas**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/api/gas)
- [**/authentication**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/authentication)
- [**/contracts/\[wallet\]/\[contractName\]/\[version\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/contracts/[wallet]/[contractName]/[version])
- [**/contracts/\[wallet\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/contracts/[wallet])
- [**/contracts/deploy/\[contractId\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/contracts/deploy/[contractId])
- [**/contracts/deploy**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/contracts/deploy)
- [**/contracts**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/contracts)
- [**/contracts/new/custom**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/contracts/new/custom)
- [**/contracts/new**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/contracts/new)
- [**/contracts/new/pre-built/\[type\]/\[contractType\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/contracts/new/pre-built/[type]/[contractType])
- [**/contracts/new/pre-built/\[type\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/contracts/new/pre-built/[type])
- [**/contracts/new/pre-built**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/contracts/new/pre-built)
- [**/contracts/release/\[contractId\]**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/contracts/release/[contractId])
- [**/contracts/release**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/contracts/release)
- [**/gas**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/gas)
- [**/**](https://thirdweb-k7myc5u9g-thirdweb.vercel.app/)

2 deleted routes:
- **/[wallet]/[network]/pack/[pack]**
- **/start**

Commit 6ad8689249cb91b798598e5302d27b387c38f66e (https://thirdweb-k7myc5u9g-thirdweb.vercel.app).

File: pages/api/gas.ts [hunk 1]
```diff
@@ -24,6 +24,10 @@ export default async (req: NextApiRequest, res: NextApiResponse) => {
       return res.status(400).json({ error: "Failed to fetch ETH price" });
     }
 
+    res.setHeader(
```
Comment (nachoiacovino) [inline pages/api/gas.ts:27]: what does this do?

File: pages/contracts/[wallet]/index.tsx [hunk 2]
```diff
@@ -119,18 +120,21 @@ const UserPageWrapped = () => {
...
     </PublisherSDKContext>
   );
-}
-
+};
 UserPage.getLayout = function getLayout(page: ReactElement) {
   return <AppLayout>{page}</AppLayout>;
 };
 
+UserPage.trackingScope = "released_contracts_profile";
```
Comment (nachoiacovino) [inline pages/contracts/[wallet]/index.tsx:134]:
```suggestion
UserPage.trackingScope = "team_profile";
```

File: pages/contracts/deploy/[contractId].tsx [hunk 3]
```diff
@@ -42,97 +39,96 @@ export default function ContractDetailPage() {
...
+    </Flex>
   );
-}
+};
 
 ContractDetailPage.getLayout = function getLayout(page: ReactElement) {
   return <AppLayout>{page}</AppLayout>;
 };
 
+ContractDetailPage.trackingScope = "contracts_deploy_detail";
```
Comment (nachoiacovino) [inline pages/contracts/deploy/[contractId].tsx:122]:
```suggestion
ContractDetailPage.trackingScope = "pre_deploy";
```

File: pages/contracts/deploy/index.tsx [hunk 4]
```diff
@@ -19,29 +15,31 @@ export default function ContractsDeployPage() {
...
+    </Flex>
   );
-}
+};
 
 ContractsDeployPage.getLayout = function getLayout(page: ReactElement) {
   return <AppLayout>{page}</AppLayout>;
 };
+
+ContractsDeployPage.trackingScope = "contracts_deploy";
```
Comment (nachoiacovino) [inline pages/contracts/deploy/index.tsx:43]:
```suggestion
ContractsDeployPage.trackingScope = "deploy_contract";
```

File: pages/contracts/release/[contractId].tsx [hunk 5]
```diff
@@ -2,51 +2,49 @@ import { Flex, Link } from "@chakra-ui/react";
...
     </PublisherSDKContext>
   );
-}
+};
 
 ContractsPublishPage.getLayout = (page: ReactElement) => (
   <AppLayout>{page}</AppLayout>
 );
+
+ContractsPublishPage.trackingScope = "release_contract_detail";
```
Comment (nachoiacovino) [inline pages/contracts/release/[contractId].tsx:48]:
```suggestion
ContractsPublishPage.trackingScope = "create_release";
```

File: pages/contracts/release/index.tsx [hunk 6]
```diff
@@ -19,32 +15,34 @@ export default function ContractsReleasePage() {
...
+    </Flex>
   );
-}
+};
 
 ContractsReleasePage.getLayout = function getLayout(page: ReactElement) {
   return <AppLayout>{page}</AppLayout>;
 };
+
+ContractsReleasePage.trackingScope = "release_contract_overview";
```
Comment (nachoiacovino) [inline pages/contracts/release/index.tsx:46]:
```suggestion
ContractsReleasePage.trackingScope = "pre_release";
```

File: pages/contracts/[wallet]/[contractName]/[version].tsx [hunk 7]
```diff
@@ -109,18 +110,22 @@ const ContractsNamePageWrapped = () => {
...
     </PublisherSDKContext>
   );
-}
+};
 
 ContractNamePage.getLayout = function getLayout(page: ReactElement) {
   return <AppLayout>{page}</AppLayout>;
 };
 
+ContractNamePage.trackingScope = "released_contract_detail";
```
Comment (nachoiacovino) [inline pages/contracts/[wallet]/[contractName]/[version].tsx:125]:
```suggestion
ContractNamePage.trackingScope = "released_contract_detail";
```

this one makes sense to me, I was calling it `specific_release` but every page is gonna be a specific release

**Review by nachoiacovino — COMMENTED** [review]
all of these are suggestions, lmk what you think

File: pages/contracts/deploy/index.tsx (hunk 4 above)
Comment (nachoiacovino) [reply pages/contracts/deploy/index.tsx:43]: deploy_custom_contract

**Comment by jnsdls** [comment]
will write a doc about / for this first

**Pull request closed without merging**